src/lib/state/index.ts

```diff
@@ -0,0 +1,4 @@
+import { Provider as EthProvider } from '@ethersproject/abstract-provider'
+import { atom } from 'jotai'
+
+export const providerAtom = atom<EthProvider | undefined>(undefined)
```

Comment:
What could we accomplish with this pattern?

Reply:
Multiple widgets sharing a single provider, which can be specified in a Context; this would allow us to support both a widgetized App (using Context), and standalone widgets (using passed provider props).

This can probably just be done by defaulting a Context-specified atom's value to that of the Context, too. That actually seems cleaner. I just wanted to mention the design need now to make sure we didn't accidentally preclude it.